src/crypto/dleq.rs

```diff
@@ -0,0 +1,1047 @@
+use std::convert::TryInto;
+
+use crate::{
+    consensus::{self, CanonicalBytes},
+    crypto,
+};
+use amplify::num::u256;
+
+use bitcoin_hashes::{self, Hash};
+
+use bitvec::{order::Lsb0, prelude::BitSlice};
+use curve25519_dalek::{
+    constants::ED25519_BASEPOINT_POINT as G, edwards::CompressedEdwardsY as ed25519PointCompressed,
+    edwards::EdwardsPoint as ed25519Point, scalar::Scalar as ed25519Scalar,
+};
+
+const ENTROPY: bool = true;
+
+fn _max_ed25519() -> u256 {
+    (u256::from(2u32) << 252) + 27742317777372353535851937790883648493u128
```

Comment:
There is a typo here: `2 = 0b0000_0010`, left shifting it by 252 ends up to `2^253 + ...`

Did some quick test:
```sage
sage: s=2**252+27742317777372353535851937790883648493
sage: s
7237005577332262213973186563042994240857116359379907606001950938285454250989
sage: s.hex()
'1000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed'
```
and the rust code
```rust
let num = (u256::from(2u32) << 252) + 27742317777372353535851937790883648493u128;
println!("{}", num);
0x2000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed
let num = (u256::from(1u32) << 252) + 27742317777372353535851937790883648493u128;
println!("{}", num);
0x1000000000000000000000000000000014def9dea2f79cd65812631a5cf5d3ed
```

Reply:
thanks! fixed.